app/graphql/resolvers/rdf.ts

```diff
@@ -314,6 +322,107 @@ export const dataCubesMetadata: NonNullable<
   );
 };
 
+export const dataCubesObservations: NonNullable<
+  QueryResolvers["dataCubesObservations"]
+> = async (_, { locale, filters }, { setup }, info) => {
+  if (filters.length > 1 && filters.some((f) => f.joinBy === undefined)) {
+    throw new Error("Can't query multiple cubes observations without joinBy!");
+  }
+
+  const { loaders, sparqlClient, cache } = await setup(info);
+  // If the cube was updated, we need to also update the filter with the correct iri.
+  const filtersWithCorrectIri: DataCubeObservationFilter[] = [];
+
+  const dataByCubeIri: Record<string, Observation[]> = {};
+  const sparqlEditorUrls: DataCubesObservations["sparqlEditorUrls"] = [];
```

Comment:
Not a fan of instantiating empty arrays/props and relying on loops and callbacks in order to fill out the values.  I think with `Promise.all()` there is a risk of partial failure or skipped steps that might make this harder to test.  I think rather having side-effects within the promise all its easier to understand if you return values and rely on functors like map, reduce, filter etc.

Reply:
Yup, good point! It would work in the case of the below `Promise.all`, however I think it would be trickier to change the approach when getting `mergedObservations`, as there we need to access the "past" value in current iteration to overwrite it "outside" of the loop we're in. I'll refactor the `Promise.all` and see if there are other approaches to deal with the second case 👍